x-pack/plugin/esql-core/src/main/java/org/elasticsearch/xpack/esql/core/expression/EmptyAttribute.java

```diff
@@ -7,22 +7,34 @@
 
 package org.elasticsearch.xpack.esql.core.expression;
 
+import org.elasticsearch.common.io.stream.StreamOutput;
 import org.elasticsearch.xpack.esql.core.tree.NodeInfo;
 import org.elasticsearch.xpack.esql.core.tree.Source;
 import org.elasticsearch.xpack.esql.core.type.DataType;
 import org.elasticsearch.xpack.esql.core.type.DataTypes;
 import org.elasticsearch.xpack.esql.core.util.StringUtils;
 
+import java.io.IOException;
+
 /**
  * Marker for optional attributes. Acting as a dummy placeholder to avoid using null
  * in the tree (which is not allowed).
  */
 public class EmptyAttribute extends Attribute {
-
     public EmptyAttribute(Source source) {
         super(source, StringUtils.EMPTY, null, null);
     }
 
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        throw new UnsupportedOperationException("doesn't escape the node");
```

Comment:
Thought: I agree with adding serialization only to things that we need to serialize.

This is probably fine, but this is something that _could_ be sent across nodes depending on optimizations. Currently, this is used in aggs when they are completely optimized away but we still need one row.

Reply:
:+1:

It isn't hard to add serialization for this one.

Reply:
I'm ok adding it when we need it. @alex-spies, how sure are you that if we added a behavior that needed to serialize this one across the nodes we'd fail a test hitting a line like this? I think our multinode tests would catch it pretty quick, but that's hard to be sure of.